lib/rules/no-assignment-of-computed-property-dependencies.js

```diff
@@ -0,0 +1,196 @@
+'use strict';
+
+const emberUtils = require('../utils/ember');
+const types = require('../utils/types');
+const decoratorUtils = require('../utils/decorators');
+const javascriptUtils = require('../utils/javascript');
+const propertySetterUtils = require('../utils/property-setter');
+const assert = require('assert');
+
+const ERROR_MESSAGE =
+  "Use `set(this, 'propertyName', 'value')` or tracked properties instead of assignment for properties that are used as computed property dependencies.";
+
+/**
+ * Gets the list of string dependent keys from a computed property.
+ *
+ * @param {Node} node - the computed property node
+ * @returns {String[]} - the list of string dependent keys from this computed property
+ */
+function getComputedPropertyDependentKeys(node) {
+  if (!node.arguments) {
+    return [];
+  }
+
+  return node.arguments
+    .filter((arg) => arg.type === 'Literal' && typeof arg.value === 'string')
+    .map((node) => node.value);
+}
+
+/**
+ * Gets a list of computed property dependency keys used inside a class.
+ *
+ * @param {Node} nodeClass - Node for the Ember class
+ * @returns {String[]} - list of dependent keys used inside the class
+ */
+function findComputedPropertyDependentKeys(nodeClass) {
+  if (types.isClassDeclaration(nodeClass)) {
+    // Native JS class.
+    return javascriptUtils.flatMap(nodeClass.body.body, (node) => {
+      const computedDecorator = decoratorUtils.findDecorator(node, 'computed');
+      if (computedDecorator) {
+        return getComputedPropertyDependentKeys(computedDecorator.expression);
+      } else {
+        return [];
+      }
+    });
+  } else if (types.isCallExpression(nodeClass)) {
+    // Classic class.
+    if (
+      nodeClass.arguments &&
+      nodeClass.arguments.length > 0 &&
+      types.isObjectExpression(nodeClass.arguments[nodeClass.arguments.length - 1])
+    ) {
+      const classObject = nodeClass.arguments[nodeClass.arguments.length - 1];
+      return javascriptUtils.flatMap(classObject.properties, (node) => {
+        if (emberUtils.isComputedProp(node.value) && node.value.arguments) {
+          return getComputedPropertyDependentKeys(node.value);
+        } else {
+          return [];
+        }
+      });
+    }
+  } else {
+    assert(false, 'Unexpected node type for an Ember class.');
+  }
+
+  return [];
+}
+
+/**
+ * Gets a list of tracked properties used inside a class.
+ *
+ * @param {Node} nodeClass - Node for the Ember class
+ * @returns {String[]} - list of tracked properties used inside the class
+ */
+function findTrackedProperties(nodeClassDeclaration) {
+  return nodeClassDeclaration.body.body
+    .filter(
+      (node) =>
+        types.isClassProperty(node) &&
+        decoratorUtils.hasDecorator(node, 'tracked') &&
```

Comment:
Should be based on imports. The following is perfectly valid:

```js
import { tracked as t } from '@glimmer/tracking';

export default class extends Service {
  @t mything;
}
```

Reply:
Updated to use imported name for `tracked` and added a test. Thanks for the reminder about this!